src/mesh/gmsh_reader.cpp

```diff
@@ -1663,16 +1685,37 @@ read_gmsh(std::string filename, int requested_grid_order)
 //        grid->output_results_vtk(9999);                    //Removed to save time (especially for large 3D grids) during flow solver test case. Grid will be outputted 'once' inside the gmsh_reader test file.
         grid->reset_initial_nodes();
 
+        //Check for periodic boundary conditions and apply
+        std::vector<dealii::GridTools::PeriodicFacePair<typename dealii::Triangulation<dim>::cell_iterator> > matched_pairs;
+        
+        if (periodic_x) {
+            dealii::GridTools::collect_periodic_faces(*grid->triangulation, x_periodic_1, x_periodic_2, 0, matched_pairs);
+        }
+
+        if (periodic_y) {
+            dealii::GridTools::collect_periodic_faces(*grid->triangulation, y_periodic_1, y_periodic_2, 1, matched_pairs);
+        }
+
+        if (periodic_z) {
+            dealii::GridTools::collect_periodic_faces(*grid->triangulation, z_periodic_1, z_periodic_2, 2, matched_pairs);
+        }
+
+        if (periodic_x || periodic_y || periodic_z) {
+            grid->triangulation->add_periodicity(matched_pairs);
+        }
+
         return grid;
+
     } else {
         return high_order_grid;
     }
 }
 
-
 #if PHILIP_DIM==1 
 #else
-template std::shared_ptr< HighOrderGrid<PHILIP_DIM, double> > read_gmsh<PHILIP_DIM,PHILIP_DIM>(std::string filename, int requested_grid_order);
+// template std::shared_ptr< HighOrderGrid<PHILIP_DIM, double> > read_gmsh<PHILIP_DIM,PHILIP_DIM>(std::string filename, const PHiLiP::Parameters::AllParameters *const parameters_input, int requested_grid_order);
```

Comment:
remove old commented code

Reply:
Fixed this in the recent commit. Removed.